src/main/scala/com/databricks/labs/overwatch/env/Database.scala

```diff
@@ -27,18 +27,54 @@ class Database(config: Config) extends SparkSessionWrapper {
   }
 
   /**
-   * register an Overwatch target table in the configured Overwatch deployment
+   * Checks if the table has optimizeWrite enabled.
+   * @param target
+   * @return
+   */
+  private def checkAutoOptimizeWrite(target: PipelineTable): Boolean = {
+    import spark.implicits._
+    val optimiseWrite: Boolean = spark.sql(s"""describe detail '${target.tableLocation}'""")
+      .select("properties").as[Map[String, String]].collect()
+      .headOption.getOrElse(throw new Exception("string"))
+      .getOrElse("delta.autoOptimize.optimizeWrite", "false")
+      .toBoolean
+    optimiseWrite
+  }
+
+
+  /**
+   * register an Overwatch target table in the configured Overwatch deployment.
+   * After release > 7.2.1 : When the tables are dropped ,it will create the table as it was previously created then check whether it has autoOptimize enabled or not.
+   * If AutoOptimize is enables then alter the table and set the property of autoOptimizeWrite to true.
```

Comment:
could you be little more explicit on the rules here. Please also note that creating a table in a way that does not match the underlying files will result in an error. If you write out the logic in English, you'll realize that the code logic does not satisfy the intent. See below.